### ISSUE TITLE
App freezes and displays fatal error when text provided in the search bar is too long

### ISSUE THREAD

**Issue by Kijek3** [laurent22/joplin#7634]
## Environment

Joplin version: 2.9.17
Platform: Windows
OS specifics: Windows 10

## Steps to reproduce

1. Click on the search bar
2. Paste very long text, line of 50000 chars and more should "work"
3. Joplin app freezes and after some time displays very long fatal error
![obraz](https://user-images.githubusercontent.com/64635454/213514874-ece00b75-a59d-4bc6-a7c2-c8c71c64a264.png)

## Describe what you expected to happen
Search performs or if it shouldn't, then app should display some message to the user that the value in search bar is too long.

## Logfile

![obraz](https://user-images.githubusercontent.com/64635454/213516258-1ab1c21c-5a3d-4b2b-a7a1-8bd863a33be8.png)

**Comment by laurent22** [comment]
Do you have any suggestions on how to replicate your bug? Should we manually type the 50000 characters from your screenshot or can you post the actual text here?

One of the first things we mention in the issue template is "DO NOT create screenshots of text".

**Comment by tomasz1986** [comment]
For the record, I've just tried to reproduce with strings that had over 1,000,000 characters, but Joplin did not crash even once.

**Comment by laurent22** [comment]
From the error message it seems to be something specific to the string he used. And that's exactly why we ask not to take screenshots of text

**Comment by Kijek3** [comment]
@laurent22 Well, application crashed for me for every long enough string that I put in the search, that's why I did not paste any text here. Also it was pretty normal chars, without any dialects in them or something sketchy. Here is my 3 test inputs, all of them are crashing application for me:
[strings.txt](https://github.com/laurent22/joplin/files/10470154/strings.txt)
1. testtest... - string used in the PR screen
2. aaa... - here you can see that every letter is transformed to the other languages equivalent, which may be helpful in investigation
3. bbb... - the simplest one, as it don't have any dialects in other languages. At first I was thinking that maybe those chars from other languages are the problem, but here you will more readable error message, like this one:
![obraz](https://user-images.githubusercontent.com/64635454/213804024-4fb023ee-a9e4-420c-af7b-86143a3134d1.png)

So it seems like when regular expression is too large, then application crashes.

Pasted message:
```
Invalid regular expression: /(^|\s|:|;|\.|,|\-|–|—|‒|_|\(|\)|\{|\}|\[|\]|!|'|"|\+|=)
(bbbbbb...)
(?=$|\s|:|;|\.|,|\-|–|—|‒|_|\(|\)|\{|\}|\[|\]|!|'|"|\+|=)/: Regular expression too large
```

EDIT: Log file: 
[log.txt](https://github.com/laurent22/joplin/files/10470247/log.txt)


@tomasz1986 Oh, that's strange, I've tested it on 2 different machines and it is always crashing application on them. One of them was low end machine and other one is like middle tier laptop. Tbh I was hoping that this bug is not connected with machine specs, but maybe it is.

**Comment by roman-r-m** [comment]
Crashed for me with the `testtesttest...` string from the file attached above.

**Comment by pedr** [comment]
It seems like the error was fixed in version 2.10.2 and beyond, I tried using the file content provided, but the error never happened on my machine.

I tested the reported version using Ubuntu 22.04 and the error was indeed happening in this older version, but I could not find the reason. I think it might be related to this commit https://github.com/laurent22/joplin/commit/a7213453c75e9959660fc022357954327612a62a

**Comment by pedr** [comment]
Today I spend more time on the issue and learned that actually, it is happening in the newest version too, but only when you try to click on the New Note or New To-Do button, meanwhile, in the reported version (`v2.9.17`) the application was crashing even if the user didn't make any actions besides pasting the input.

About the error message, I found it was a bug that was fixed in Chrome 95/96, which is from early 2022. From what I researched it seems like Electron uses a newer version and Node.js used in the project also uses a V8 version that should have this fixed, but maybe I'm wrong, I'm still getting used to the Electron/Desktop environment 
https://bugs.chromium.org/p/chromium/issues/detail?id=1264014

Considering that is not the problem I started investigating the problem further.

It seems to only happen when you try to click on the New note or New to-do button beside the input field, but only if you click on it "right away" (right away being some seconds in my machine), I guess the program is still processing the search and when it tries to create a new note it finds some undefined state, I'm still investigating to narrow down when the problem happens.
